using_restricted_array.rb

```diff
@@ -8,23 +8,52 @@
 
 ## Calculates the length of the restricted integer array_size
 def length(array)
-  puts "NOT IMPLEMENTED"
+  count = 0
+  continue = true
+  while continue
+    array[count] != nil ? count += 1 : continue = false
+  end
+  return count
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  index = 0
+  continue = true
+  while continue
+    if array[index] != nil
+      puts array[index]
+      index += 1
+    else
+      continue = false
+    end
+  end
 end
 
 # Reverses the values in the integer array
 def reverse(array, length) # Ruby
-  puts "NOT IMPLEMENTED"
+  i = 0
+  j = length
+  length/2.times do
+    temp = array[i]
+    array[i] = array[j]
+    array[j] = temp
+    i += 1
+    j -= 1
+  end
+  return array
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+  has_found = false
+  length.times do |i|
+    if array[i] == value_to_find
+      has_found = true
+    end
+  end
+  return has_found
 end
 
 # Sorts the array in ascending order.

```

Comment:
See https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/classroom/Sorting.md for more learning on this not yet implemented method.